plugins/connectors/src/flyteplugins/connectors/databricks/connector.py

```diff
@@ -0,0 +1,150 @@
+import http
+import json
+import os
+import typing
+from dataclasses import dataclass
+from typing import Optional
+
+import aiohttp
+from flyte import logger
+from flyte.connectors import AsyncConnector, ConnectorRegistry, Resource, ResourceMeta
+from flyte.connectors.utils import convert_to_flyte_phase
+from flyteidl2.core.execution_pb2 import TaskExecution, TaskLog
+from flyteidl2.core.tasks_pb2 import TaskTemplate
+from google.protobuf.json_format import MessageToDict
+
+DATABRICKS_API_ENDPOINT = "/api/2.1/jobs"
+DEFAULT_DATABRICKS_INSTANCE_ENV_KEY = "FLYTE_DATABRICKS_INSTANCE"
+
+
+@dataclass
+class DatabricksJobMetadata(ResourceMeta):
+    databricks_instance: str
+    run_id: str
+
+
+def _get_databricks_job_spec(task_template: TaskTemplate) -> dict:
+    custom = MessageToDict(task_template.custom)
+    container = task_template.container
+    envs = task_template.container.env
+    databricks_job = custom.get("databricksConf")
+    if databricks_job is None:
+        raise ValueError("Missing Databricks job configuration in task template.")
+    if databricks_job.get("existing_cluster_id") is None:
+        new_cluster = databricks_job.get("new_cluster")
+        if new_cluster is None:
+            raise ValueError("Either existing_cluster_id or new_cluster must be specified")
+        if not new_cluster.get("docker_image"):
+            new_cluster["docker_image"] = {"url": container.image}
+        if not new_cluster.get("spark_conf"):
+            new_cluster["spark_conf"] = custom.get("sparkConf", {})
+        if not new_cluster.get("spark_env_vars"):
+            new_cluster["spark_env_vars"] = {env.key: env.value for env in envs}
+        else:
+            new_cluster["spark_env_vars"].update({env.key: env.value for env in envs})
+    # https://docs.databricks.com/api/workspace/jobs/submit
+    databricks_job["spark_python_task"] = {
+        "python_file": "plugins/connectors/src/flyteplugins/connectors/databricks/entrypoint.py",
+        "parameters": list(container.args),
+        "source": "GIT",
+    }
+    databricks_job["git_source"] = {
+        "git_url": "https://github.com/flyteorg/flyte-sdk",
+        "git_provider": "gitHub",
+        # https://github.com/flyteorg/flyte-sdk/tree/6f4b65af95f79a4bc1046e7ffa6d0cc1ae61177c
+        "git_commit": "6f4b65af95f79a4bc1046e7ffa6d0cc1ae61177c",
```

Comment:
? why the hardcoding? Why not the git release tag then, which can be the same as the version?

Reply:
i can change that to main